SRC/CLASSES.CPP

```diff
@@ -55,13 +55,42 @@ void Message_board::add_message(char msg[])
 }
 
 void Message_board::draw(){
- int a,yoffs=5;
+ int a,b,yoffs=5,neutral_color=32;
 
  if (KILLING_MODE==DEATHMATCH)  yoffs=30;
 
  if (time_count>0) time_count--; else clear();
  for (a=0;a<messages;a++)
-  writefonts(5,yoffs+a*9,message[a],32);
+ {
+   char copy[MAX_MESSAGE_LENGTH];
+   strcpy(copy,message[a]);
+   char* split = strchr(copy,':');
+   if (split && strlen(split) > 1)
+   {
+     const int y_offset = yoffs+a*9;
+     int player_color = neutral_color;
+     const char stored_char = *(split + 1);
+     *(split+1) = '\0'; // Split text to "<name>:" first
+
+     for(b=0;b<MAX_PLAYERS;b++)
+     {
+       if (player[b].enabled && !strncmp(player[b].name, copy, strlen(copy)-1))
```

Comment:
`strlen(copy) - 1` -> `split - copy`?

Reply:
That'll do! [Updated](https://github.com/suomipelit/ultimatetapankaikki/pull/172/commits/e5dda1fa4d1bcda904b3969cd6d19b1ef2bcd4bc).

Also increased message size a bit (while already incompatible network changes since v2), optimized drawing (one less dry write to figure out name text length) and some other minor changes.